src/ChainTransactional.php

```diff
@@ -66,7 +66,7 @@ public function commit(): void
         $mustRollback = false;
         $firstException = null;
 
-        while ($layer = \array_shift($layers)) {
+        while ($layer = \array_pop($layers)) {
```

Comment:
1. You change logic only for `commit`, but in `rollback` we must use reverse logic too.
2. I don't see test where we check correct calls with priorities.